tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java

```diff
@@ -149,6 +151,27 @@ public void initDebugSession(DebugUtils.DebuggeeExecutionKind executionKind) thr
         initDebugSession(executionKind, launchConfigs);
     }
 
+
+    /**
+     * Initialize test debug session.
+     *
+     * @param executionKind Defines ballerina command type to be used to launch the debuggee.(If set to null, adapter
+     *                      will try to attach to the debuggee, instead of launching)
+     * @throws BallerinaTestException if any exception is occurred during initialization.
+     */
+    public Boolean initDebugSessionInTerminal(DebugUtils.DebuggeeExecutionKind executionKind, String terminalKind)
+            throws BallerinaTestException {
+        HashMap<String, Object> launchConfigs = new HashMap<>();
+        HashMap<String, Object> extendedCapabilities = new HashMap<>();
+        extendedCapabilities.put("supportsReadOnlyEditors", true);
+        launchConfigs.put("capabilities", extendedCapabilities);
+        launchConfigs.put("terminal", terminalKind);
+        initDebugSession(executionKind, launchConfigs);
+
+        return debugClientConnector.getRequestManager().getDidRunInIntegratedTerminal();
+    }
```

Comment:
Can we reuse this constructor within the other one (`initDebugSession(DebugUtils.DebuggeeExecutionKind executionKind)`) to get rid of the duplicated code? WDYT?

Reply:
We can set the terminal kind to an empty string, and pass on the debug session initialization to this, thereby avoiding the code duplication.

Reply:
Yeah thats what I meant. +1